src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs

```diff
@@ -152,6 +152,12 @@ public static class WebScriptHostBuilderExtension
                     services.AddSingleton<IFileMonitoringService, FileMonitoringService>();
                     services.TryAddEnumerable(ServiceDescriptor.Singleton<IHostedService, IFileMonitoringService>(p => p.GetService<IFileMonitoringService>()));
 
+                    IHostJsonConfigProvider hostJsonConfigProvider = rootServiceProvider.GetService<IHostJsonConfigProvider>();
+                    if (hostJsonConfigProvider != null)
```

Comment:
With HostOptionsProvider a ScriptHost level service, we won't have to do any of this RegisterOptionTypes anymore.

Reply:
Removed